lib/ansible/template/__init__.py

```diff
@@ -252,7 +252,19 @@ def _unroll_iterator(func):
     explicitly use ``|list`` to unroll.
     """
     def wrapper(*args, **kwargs):
-        ret = func(*args, **kwargs)
+        try:
+            ret = func(*args, **kwargs)
+        except (UndefinedError, AnsibleUndefinedVariable):
+            raise
+        except Exception as e:
+            if 'undefined' in to_native(e):
+                from jinja2.runtime import Undefined
+                undefined_input = any(isinstance(arg, (Undefined, AnsibleUndefined)) for arg in args)
```

Comment:
Isn't checking for `Undefined` enough since the inheritance is `Undefined` -> `StrictUndefined` -> `AnsibleUndefined`?